grafana-plugin/src/components/ExtensionLinkMenu/ExtensionLinkMenu.tsx

```diff
@@ -0,0 +1,131 @@
+            {renderItems(categorised[category])}
+          </Menu.Group>
+        ))}
+        {showDivider && <Menu.Divider key="divider" />}
+        {renderItems(uncategorised)}
+      </>
+    </Menu>
+  );
+}
+
+// This menu item is a temporary workaround for the fact that the Incident plugin doesn't
+// register its own extension link.
+// TODO: remove this once Incident is definitely registering its own extension link.
+function DeclareIncidentMenuItem({ extensions, declareIncidentLink, grafanaIncidentId }: Props): ReactElement | null {
+  const declareIncidentExtensionLink = extensions.find(
+    (extension) => extension.pluginId === 'grafana-incident-app' && extension.title === 'Declare incident'
+  );
+  if (
+    // Don't show a custom Declare incident button if the Grafana Incident plugin already configured one.
+    declareIncidentExtensionLink !== undefined ||
+    // Don't show a custom Declare incident button if there's no valid link.
+    declareIncidentLink === undefined ||
+    // Don't show the button if an incident has already been declared from this alert group.
+    grafanaIncidentId !== null
+  ) {
+    return null;
+  }
+  return (
+    <PluginBridge plugin={SupportedPlugin.Incident}>
+      <Menu.Group key={'Declare incident'} label={'Incident'}>
+        {renderItems([
+          {
+            type: 'link',
+            path: declareIncidentLink,
+            icon: 'fire',
+            category: 'Incident',
+            title: 'Declare incident',
+            pluginId: 'grafana-oncall-app',
+          },
+        ])}
+      </Menu.Group>
+    </PluginBridge>
+  );
+}
+
+function renderItems(extensions: PluginExtensionLink[]): JSX.Element[] {
+  return extensions.map((extension) => (
+    <Menu.Item
+      ariaLabel={extension.title}
+      icon={extension?.icon || 'plug'}
+      key={extension.id}
+      label={truncateTitle(extension.title, 25)}
+      onClick={(event) => {
+        if (extension.path) {
+          return void global.open(locationUtil.assureBaseUrl(extension.path), '_blank');
+        }
+        extension.onClick?.(event);
+      }}
+    />
+  ));
+}
+
+type ExtensionLinksResult = {
+  uncategorised: PluginExtensionLink[];
+  categorised: Record<string, PluginExtensionLink[]>;
+};
+
+function useExtensionLinksByCategory(extensions: PluginExtensionLink[]): ExtensionLinksResult {
+  return useMemo(() => {
+    const uncategorised: PluginExtensionLink[] = [];
+    const categorised: Record<string, PluginExtensionLink[]> = {};
+
+    for (const link of extensions) {
+      if (!link.category) {
+        uncategorised.push(link);
+        continue;
+      }
+
+      if (!Array.isArray(categorised[link.category])) {
+        categorised[link.category] = [];
+      }
+      categorised[link.category].push(link);
+      continue;
+    }
+
+    return {
+      uncategorised,
+      categorised,
+    };
+  }, [extensions]);
+}
+
+function truncateTitle(title: string, length: number): string {
```

Comment:
This should ideally live in some common place, e.g. we can create new file `src/utils/string` and put it there

Reply:
Done!